core/arch/aarch64/emit_utils.c

```diff
@@ -205,24 +272,76 @@ exit_cti_reaches_target(dcontext_t *dcontext, fragment_t *f, linkstub_t *l,
 }
 
 void
-patch_stub(fragment_t *f, cache_pc stub_pc, cache_pc target_pc, bool hot_patch)
+patch_stub(fragment_t *f, cache_pc stub_pc, cache_pc target_pc, cache_pc target_prefix_pc,
+           bool hot_patch)
 {
+    /* Compute offset as unsigned, modulo arithmetic. */
+    ptr_uint_t off = (ptr_uint_t)target_pc - (ptr_uint_t)stub_pc;
+    if (off + 0x8000000 < 0x10000000) {
+        /* target_pc is a near fragment. We can get there with a B
+         * (OP_b, 26-bit signed immediate offset).
+         * i#1911: Patching arbitrary instructions to an unconditional branch
+         * is theoretically not sound. Architectural specifications do not
+         * guarantee safe behaviour or any bound on when the change will be
+         * visible to other process elements.
+         */
+        *(uint *)vmcode_get_writable_addr(stub_pc) =
+            (0x14000000 | (0x03ffffff & off >> 2));
+        if (hot_patch)
+            machine_cache_sync(stub_pc, stub_pc + 4, true);
+        return;
+    }
+    /* target_pc is a far fragment. We must use an indirect branch. Note that the indirect
+     * branch needs to be to the fragment prefix, as we need to restore the clobbered
+     * regs. */
```

Comment:
nit: prefer `*/` on own line (though this may be a losing battle since there's no clang-format rule to enforce it...could be convinced to remove the rule)

Reply:
I made the change. `*/` on its own line looks better somehow.